### PR TITLE
rng_hotplug: Modify plug and unplug method to adjust to pcie bus

### PR DESCRIPTION
**Pull request by lijinlijin**
modify hotplug and hot-unplug method to support rng device plug and unplug under pcie bus.

ID:1570472

Signed-off-by: Li Jin <lijin@redhat.com>

### CODE REVIEW

File: qemu/tests/rng_hotplug.py [hunk 1]
```diff
@@ -37,26 +37,16 @@ def get_rng_id(vm):
...
-        time.sleep(5)
-
-        error_context.context("Check %s from qtree after hotplug" % dev,
-                              logging.info)
-        qtree_output = dev.verify_hotplug(output, vm.monitor)
-        if not qtree_output:
+        out, ver_out = vm.devices.simple_hotplug(dev, vm.monitor)
+        if not ver_out:
             msg = "no % device in qtree after hotplug" % dev
             raise exceptions.TestFail(msg)
```
Comment (luckyh) [inline qemu/tests/rng_hotplug.py]: Better to emit `output` in error message.

File: qemu/tests/rng_hotplug.py [hunk 2]
```diff
@@ -37,26 +37,16 @@ def get_rng_id(vm):
...
-        time.sleep(5)
-
-        error_context.context("Check %s from qtree after unplug" % dev,
-                              logging.info)
-        qtree_output = dev.verify_unplug(output, vm.monitor)
-        if not qtree_output:
+        out, ver_out = vm.devices.simple_unplug(dev, vm.monitor)
+        if not ver_out:
             msg = "Still get %s in qtree after unplug" % dev
             raise exceptions.TestFail(msg)
```
Comment (luckyh) [inline qemu/tests/rng_hotplug.py]: The same as the above.

**Comment by lijinlijin** [comment]
All cases passed on both rhel and windows guests:

Q35:
'01-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35' Finshed 2018-09-27 03:55:37 2018-09-27 04:11:28 PASS 951.094116
'02-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rh_kernel_update.q35' Finshed 2018-09-27 04:11:28 2018-09-27 04:12:12 ERROR 43.87114
'03-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35' Finshed 2018-09-27 04:12:12 2018-09-27 04:30:08 PASS 1076.08487
'04-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.multi_rngs.q35' Finshed 2018-09-27 04:30:08 2018-09-27 04:31:58 PASS 109.280987
'05-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.q35' Finshed 2018-09-27 04:31:58 2018-09-27 04:33:06 PASS 68.252936
'06-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_reboot.q35' Finshed 2018-09-27 04:33:06 2018-09-27 04:35:19 PASS 133.078378
'07-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_system_reset.q35' Finshed 2018-09-27 04:35:19 2018-09-27 04:37:31 PASS 131.75222
'08-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_unplug.q35' Finshed 2018-09-27 04:37:31 2018-09-27 04:38:34 PASS 63.244295
'09-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_plug.q35' Finshed 2018-09-27 04:38:34 2018-09-27 04:39:37 PASS 62.958002
'10-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.repeat_in_loop.q35' Finshed 2018-09-27 04:39:37 2018-09-27 05:23:48 PASS 2650.670436
'11-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.multi_rngs.q35' Finshed 2018-09-27 05:23:48 2018-09-27 05:28:10 PASS 262.016281
'12-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.q35' Finshed 2018-09-27 05:28:10 2018-09-27 05:31:00 PASS 170.246087
'13-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_reboot.q35' Finshed 2018-09-27 05:31:00 2018-09-27 05:35:54 PASS 294.376777
'14-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_system_reset.q35' Finshed 2018-09-27 05:35:54 2018-09-27 05:41:27 PASS 332.99114
'15-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_unplug.q35' Finshed 2018-09-27 05:41:27 2018-09-27 05:44:10 PASS 162.769797
'16-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_plug.q35' Finshed 2018-09-27 05:44:10 2018-09-27 05:46:20 PASS 129.371497
'17-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.rng_hotplug.repeat_in_loop.q35' Finshed 2018-09-27 05:46:20 2018-09-27 09:38:50 PASS 13949.904768

PC:
'01-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads' Finshed 2018-09-27 05:01:29 2018-09-27 05:15:08 PASS 819.64645
'02-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rh_kernel_update' Finshed 2018-09-27 05:15:08 2018-09-27 05:15:34 ERROR 25.969327
'03-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads' Finshed 2018-09-27 05:15:34 2018-09-27 05:27:13 PASS 698.398164
'04-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.multi_rngs' Finshed 2018-09-27 05:27:13 2018-09-27 05:34:31 PASS 438.566343
'05-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng' Finshed 2018-09-27 05:34:31 2018-09-27 05:35:20 PASS 48.222086
'06-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_reboot' Finshed 2018-09-27 05:35:20 2018-09-27 05:37:01 PASS 101.295298
'07-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_system_reset' Finshed 2018-09-27 05:37:01 2018-09-27 05:38:58 PASS 116.661254
'08-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_unplug' Finshed 2018-09-27 05:38:58 2018-09-27 05:39:46 PASS 48.753124
'09-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_plug' Finshed 2018-09-27 05:39:46 2018-09-27 05:40:27 PASS 40.35113
'10-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.rng_hotplug.repeat_in_loop' Finshed 2018-09-27 05:40:27 2018-09-27 06:16:26 PASS 2158.990958
'11-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.multi_rngs' Finshed 2018-09-27 06:16:26 2018-09-27 06:19:11 PASS 164.630853
'12-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng' Finshed 2018-09-27 06:19:11 2018-09-27 06:20:59 PASS 108.842876
'13-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_reboot' Finshed 2018-09-27 06:20:59 2018-09-27 06:24:09 PASS 189.343214
'14-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_system_reset' Finshed 2018-09-27 06:24:09 2018-09-27 06:28:19 PASS 249.787435
'15-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_unplug' Finshed 2018-09-27 06:28:19 2018-09-27 06:29:55 PASS 96.915577
'16-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.one_rng.with_shutdown_after_plug' Finshed 2018-09-27 06:29:55 2018-09-27 06:31:28 PASS 92.141457
'17-Host_RHEL.m8.u0.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.rng_hotplug.repeat_in_loop' Finshed 2018-09-27 06:31:28 2018-09-27 09:47:41 PASS 11773.106877

File: qemu/tests/rng_hotplug.py (hunk 1 above)
Comment (vivianQizhu) [reply qemu/tests/rng_hotplug.py]: Why not use `test.fail()`?

Comment (vivianQizhu) [reply qemu/tests/rng_hotplug.py]: Okay ignore me, that's nothing to do with this pull request.

File: qemu/tests/rng_hotplug.py (hunk 2 above)
Comment (lijinlijin) [reply qemu/tests/rng_hotplug.py]: if plug/unplug fails, test will be interrupted by virttest.qemu_monitor.QMPCmdError, and error msg will be shown there. No chance to show here at all.

@luckyh If you think out is useless here, I can modify to:
...
out = vm.devices.simple_unplug(dev, vm.monitor)
if not out[1]:
...

Comment (luckyh) [reply qemu/tests/rng_hotplug.py]: ok, makes sense, thanks.

**Pull request merged**